src/reactviews/pages/ConnectionDialog/components/fabricWorkspaceViewer.tsx

```diff
@@ -0,0 +1,350 @@
+                            />
+                            <Text truncate className={styles.hideTextOverflowCell}>
+                                {item.displayName}
+                            </Text>
+                        </div>
+                    </DataGridCell>
+                ),
+            }),
+            createTableColumn<FabricSqlGridItem>({
+                columnId: "type",
+                renderHeaderCell: () => `${Loc.connectionDialog.typeColumnHeader}`,
+                renderCell: (item) => (
+                    <DataGridCell>
+                        <Text truncate className={styles.hideTextOverflowCell}>
+                            {item.typeDisplayName}
+                        </Text>
+                    </DataGridCell>
+                ),
+            }),
+        ],
+        [],
+    );
+
+    function handleServerSelected(database: FabricSqlGridItem) {
+        setSelectedRowId(database.id);
+        onSelectDatabase(database);
+    }
+
+    //#endregion Hooks
+
+    //#region Render helper methods
+
+    const renderGridContent = (): React.ReactNode => {
+        // Workspace list states
+        if (fabricWorkspacesLoadStatus.status === ApiStatus.NotStarted) {
+            return undefined;
+        }
+
+        if (fabricWorkspacesLoadStatus.status === ApiStatus.Loading) {
+            return renderLoadingWorkspaces();
+        }
+
+        if (fabricWorkspacesLoadStatus.status === ApiStatus.Error) {
+            return renderWorkspacesError();
+        }
+
+        if (fabricWorkspacesLoadStatus.status === ApiStatus.Loaded && !selectedWorkspace) {
+            return renderNoSelectedWorkspace(fabricWorkspacesLoadStatus.message);
+        }
+
+        // Workspace selection states
+        if (!selectedWorkspace || selectedWorkspace.loadStatus.status === ApiStatus.NotStarted) {
+            // should only hit this state between renders while the state settles
+            return undefined;
+        }
+
+        if (selectedWorkspace.loadStatus.status === ApiStatus.Loading) {
+            return renderLoadingDatabases();
+        }
+
+        if (selectedWorkspace.loadStatus.status === ApiStatus.Error) {
+            return renderDatabasesError();
+        }
+
+        if (selectedWorkspace.loadStatus.status === ApiStatus.Loaded) {
+            if (databasesForSelectedWorkspace.length === 0) {
+                return renderNoDatabasesFound();
+            } else {
+                return renderDataGrid();
+            }
+        }
+
+        return (
+            <>
+                <Label>Unexpected state:</Label>
+                <Label>FabricWorkspaceLoadStatus: {fabricWorkspacesLoadStatus.status}</Label>
+                <Label>SelectedWorkspace: {selectedWorkspace?.id}</Label>
+                <Label>SelectedWorkspace.Status: {selectedWorkspace?.loadStatus.status}</Label>
+                <Label>SelectedWorkspace.Databases: {selectedWorkspace?.databases.length}</Label>
```

Comment:
Localize labels on lines 184 to 188

Reply:
No need because it's debuging info for a state that should never be hit.  I'll add a comment.